EasyCommands/BlockHandlers/BlockHandlers.cs

```diff
@@ -71,9 +76,29 @@ public static class BlockHandlerRegistry {
                 return blockHandlers[blockType];
             }
 
-            public static List<Object> GetBlocks(Block blockType, Func<IMyTerminalBlock, bool> selector) => blockHandlers[blockType].GetBlocks(selector);
+            public static void invalidateCaches() {
+                PROGRAM.invalidBlockCache = true;
+                PROGRAM.invalidGroupCache = true;
+            }
+
+            public static List<Object> GetBlocks(Block blockType, Func<IMyTerminalBlock, bool> selector = null) {
+                if (PROGRAM.invalidBlockCache) {
+                    PROGRAM.GridTerminalSystem.GetBlocks(PROGRAM.blockCache);
+                    PROGRAM.invalidBlockCache = false;
+                }
+                return blockHandlers[blockType].SelectBlocks(PROGRAM.blockCache, selector);
+            }
+            public static List<Object> GetBlocks(Block blockType, string selectorString) => GetBlocks(blockType, b => b.CustomName.Equals(selectorString));
```

Comment:
nit: I think we only use this for one call.  I suggest we inline and just use the lambda:
```b => b.CustomName.Equals(selectorString)```

Saves a decent chunk of chars

Reply:
that function should have served as the hook in point for memoization. But since i have no idea how to do that efficitantly right now it is not needed